compose: Extracting "compose_validate.js".

Signior-X opened the pull request:
This commit first moves the compose.validate() function out
with the functions that are needed by it. Then one by one
checked for which function is now not needed in compose.js.

This moves all validation related functions out of "compose.js"
to "compose_validate.js".

<!-- What's this PR for?  (Just a link to an issue is fine.) -->

**Testing plan:** <!-- How have you tested? -->
Tested this with `tools/lint` and `tools/test-js-with-node`


<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

@showell @timabbott FYI

Signior-X commented:
I tested the coverage.
Before and after
![image](https://user-images.githubusercontent.com/56730716/123442152-3866df80-d5f2-11eb-90f7-810c2e3c679c.png)

andersk commented:
You’ve changed the order of the extracted functions relative to each other. Was that intentional? If not, it would be better to leave them in the same order so that this is easier to review with tools like `git show --color-moved`.

Signior-X commented:
Yes thanks to notify this @andersk 
I have updated the PR keeping the functions in the same order.

timabbott commented:
I don't understand the purpose of the `compose_announce` module.  I'd have expected those functions to just be moved to `compose_validate.js` in the original commit, just like you did in the third commit.  And I believe with that change, you could squash this and just have a single commit that moves a bunch of code without modification to `compose_validate.js` (with very few changes to individual lines).

Most of what's messy about the first commit right now is trying to access properties that were left behind in `compose.js`.

Signior-X commented on frontend_tests/node_tests/compose.js:
```diff
@@ -112,58 +111,6 @@ function initialize_handlers({override}) {
     compose.initialize();
 }
 
-test_ui("test_wildcard_mention_allowed", () => {
```
Moved this test to `node_tests/compose_validate.js` as all wildcard mentions right now are used by it. So, testing there will be better.

Signior-X commented:
@timabbott @showell I have updated and squashed the commits. This now don't have any cyclic dependency where `compose.js` is used in `compose_validate.js`.
Also all the functions are as it is copied, changing the places where they were used and changing the function for resetting `user_acknowledged_all_everyone` to explicitly set the values.

timabbott commented:
Merged, thanks @Signior-X!  I think this version is a lot cleaner than the earlier ones.
